docs: Move more sections under hardware integration

caksoylar opened the pull request:
Move portions of pages that describe hardware integration parts from feature pages into dedicated pages under Hardware Integration section.

- Defining battery sensing for battery levels
- Underglow and backlight (discussed in https://github.com/zmkfirmware/zmk/pull/2485#pullrequestreview-2327791146)
  - I decided to have a Lighting category index page and two subpages. The index page again emphasizes when to use each system
  - Although I added links to pin control page in both pages, this is (for now) a superficial integration and further refactoring to reduce duplication should be done in another PR

## PR check-list

- [x] Branch has a [clean commit history](https://zmk.dev/docs/development/contributing/pull-requests#clean-commit-history)
- [x] [Pre-commit](https://zmk.dev/docs/development/local-toolchain/pre-commit) used to check formatting of files, commit messages, etc.

Nick-Munnich commented:
I agree that the additional refactoring could be done in a future PR. I think that the config and feature pages ought to also be combined into one. Battery bit LGTM.

caksoylar commented:
> I think that the config and feature pages ought to also be combined into one. 

How do you envision this? I thought it's nice to have all the features' config as reference under configuration. If we move them to e.g. the feature page then the config reference section will be incomplete.

Nick-Munnich commented:
> > I think that the config and feature pages ought to also be combined into one.
> 
> How do you envision this? I thought it's nice to have all the features' config as reference under configuration. If we move them to e.g. the feature page then the config reference section will be incomplete.

I was unclear I think: feature pages combined into one and config pages combined into one, making two from four.

caksoylar commented:
Got it, yeah that makes sense. Will break quite a few links, and I should remember to add redirects.

caksoylar commented:
I (finally) consolidated the underglow+backlight feature and config pages as well. I added redirects for the old pages, though I don't think they work on local builds or in deploy previews.

While debugging any missing links, I also noticed the missing anchors on the dongle page, opened #2764.

Another extra is I removed an incorrect redirect from `docs/features/encoders` to the keymap section, since IIRC we didn't end up moving that page. I included that in this PR since I am already touching redirects and I couldn't cleanly cherry-pick it.

Nick-Munnich commented on docs/docs/intro.md:
```diff
@@ -28,8 +28,8 @@ ZMK is currently missing some features found in other popular firmware. This tab
 | [Display Support](features/displays.md)[^1]                                                                                                                | 🚧  |    🚧     | ✅  |
-| [RGB Underglow](features/underglow.md)                                                                                                                     | ✅  |    ✅     | ✅  |
-| [Backlight](features/backlight.mdx)                                                                                                                        | ✅  |    ✅     | ✅  |
+| [RGB Underglow](features/lighting.md#rgb-underglow)                                                                                                        | ✅  |    ✅     | ✅  |
+| [Backlight](features/lighting.md#backlight)                                                                                                                | ✅  |    ✅     | ✅  |
```
Thoughts on consolidating this to "LED Lighting" or something similar?

caksoylar replied:
I can see that, maybe "LED-based Lighting"?

caksoylar replied:
Changed now, thanks!

Nick-Munnich reviewed (approved):
LGTM overall, just one little note.
